docs/versionmigration.md

```diff
@@ -0,0 +1,455 @@
+    stream_received.timeseries.on_data_received = on_data_releasedorreceived_handler
+    buffer.on_data_released = on_data_releasedorreceived_handler
+    stream_received.timeseries.on_raw_received = on_rawdata_releasedorreceived_handler
+    buffer.on_raw_released = on_rawdata_releasedorreceived_handler
+    new_stream.events.on_data_received = on_event_data_handler
+    # metadata callback assignments
+    stream_received.on_stream_closed = on_stream_closed_handler
+    stream_received.properties.on_changed = on_stream_properties_changed_handler
+    new_stream.timeseries.on_definitions_changed = on_parameter_definitions_changed_handler
+    new_stream.events.on_definitions_changed = on_event_definitions_changed_handler
+    new_stream.on_package_received = on_package_received_handler
+
+input_topic.on_stream_received = on_stream_received_handler
+
+# Note that these could be in a different file completely, defined by other classes, having access to all context of the stream and topic it is for
+def on_dataframe_received_handler(stream: StreamConsumer, data: pandas.DataFrame):
+    # do great things
+    stream.topic.commit() # or any other topic method/property
+
+def on_data_releasedorreceived_handler(stream: StreamConsumer, data: TimeseriesData):
+    pass
+    
+def on_rawdata_releasedorreceived_handler(stream: StreamConsumer, data: TimeseriesDataRaw):
+    pass
+
+def on_event_data_handler(stream: StreamConsumer, data: EventData):
+    pass
+
+def on_stream_closed_handler(stream: StreamConsumer, end_type: StreamEndType):
+    pass 
+
+def on_stream_properties_changed_handler(stream: StreamConsumer):
+    pass
+
+def on_parameter_definitions_changed_handler(stream: StreamConsumer):
+    pass
+
+def on_event_definitions_changed_handler(stream: StreamConsumer):
+    pass
+
+def on_package_received_handler(stream: StreamConsumer, package: StreamPackage):
+    pass
+
+… the rest of your code
+```
+
+### 'with' statement should be used with some classes in python
+
+Certain classes now use unmanaged resources, and to prevent memory leaks, we have incorporated the Python 'with' syntax for resource management.
+
+These are:
+- EventData: important to be disposed whenever manually created or received in callbacks
+- TimeseriesData: important to be disposed whenever manually created or received in callbacks
+- TimeseriesDataRaw: important to be disposed whenever manually created or received in callbacks
+- StreamPackage: important to be disposed whenever manually created or received in callbacks
+- StreamConsumer: also supports `dispose()` and automatically disposes when stream is closed.
+- StreamProducer: also supports `dispose()` and automatically disposes when stream is closed.
+- TopicConsumer: unless you're frequently subscribing to topics, this is not something you have to be too concerned about.
+- TopicProducer: unless you're frequently subscribing to topics, this is not something you have to be too concerned about.
```

Comment:
these are not showing as a bullet point list:
![image](https://user-images.githubusercontent.com/50139826/221959107-6d698f8e-8229-4bd2-9985-c25c949b3974.png)

Reply:
Checking. in readme looks fine, but not in generated docs.

Reply:
resolved
![image](https://user-images.githubusercontent.com/79838809/221963809-5216d287-2caa-4019-8f30-b87e5f808125.png)